### PR TITLE
#170947565 mark all notifications as read

### PR DESCRIPTION
**Pull request by Ntare22**
#### What does this PR do?
This feature allows user to mark all notifications as read
#### Description of Task to be completed?
- add the function in middleware to check for unread notfications
- created notification controller
- created route for `read-all`
- created seeders to test the functionalit
- added tests for the functions
- added swagger documentation
#### How should this be manually tested?
- clone this repo
- setup database according to the readme
- login as seeded user `jdev@andela.com` with password `Bien@BAR789`
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947565](https://www.pivotaltracker.com/story/show/170947565)
#### Screenshots (if appropriate)
image
image
image
#### Questions:

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+<body>
+    <div class='notifs-container'>
+        <div id='notif-title' >
+            <span>NOTIFICATIONS</span>
+        </div>
+        <div class="notifications"></div>
+    </div>
+    <script src="/socket.io/socket.io.js"></script>
+    <script>
+        const socket = io('http://localhost:8000');
```
Comment (izzett222) [inline public/index.html]: hey Jim could we use the base_URL env variable cause on Heroku that link won't work

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        socket.on('notification', (data) => {
+            console.log(data, 'wow!!!!!');
```
Comment (izzett222) [inline public/index.html]: could you remove the console

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            socket.emit('messageFromClient', 'This is from the client')
+        });
+    </script>
+</body>
+</html>
```
Comment (izzett222) [inline public/index.html]: could you add a new line on this line
```suggestion
</html>

```

File: src/models/notifications.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  const Notifications = sequelize.define('Notifications', {
+    status: DataTypes.ENUM('read, unread'),
+    content: DataTypes.STRING,
+    recieverEmail: DataTypes.STRING,
+    recieverId: DataTypes.STRING
+  }, {});
+  Notifications.associate = () => {
+    // Notifications.belongsTo(models.User, {
+    //   foreignKey: 'recieverEmail',
+    //   as: 'user',
```
Comment (Mnickii) [inline src/models/notifications.js]: remove commented code

File: src/index.js [hunk 5]
```diff
@@ -22,6 +24,8 @@ i18n.configure({
 });
 
 const app = express();
+// const http = require('http').createServer(app);
```
Comment (izzett222) [inline src/index.js]: could you remove the commented code

**Review by izzett222 — CHANGES_REQUESTED** [review]

**Review by Mnickii — DISMISSED** [review]
LGTM

**Review by izzett222 — APPROVED** [review]

**Review by bbaime98 — APPROVED** [review]

**Review by Baraka-Mugisha — APPROVED** [review]

**Pull request merged**